### PR TITLE
[quant][pt2e][refactor] Remove `backend_config` from `_maybe_insert_input_observers_for_node`

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #98094

Summary:
The goal is to remove the need to use backend_config when pt2e flow code call this function

Test Plan:
python test/test_quantization.py TestQuantizeFx

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by vkuzo** [comment]
> The goal is to remove the need to use backend_config when pt2e flow code call this function

can you share some context on why?

**Comment by jerryzh168** [comment]
> > The goal is to remove the need to use backend_config when pt2e flow code call this function
> 
> can you share some context on why?

sure, we don't need this in prepare since validation should be done in quantizer.validate(...) function, so even we have this object (or something similar), it should live in quantizer instead

File: torch/ao/quantization/fx/utils.py [hunk 1]
```diff
@@ -86,25 +86,23 @@ class ObservedGraphModuleAttrs:
...
-        if bias_index is not None and bias_index < len(node.args) and node.args[bias_index] is arg:
-            return True
-        return node.kwargs.get("bias") is arg
-    return False
+    bias_index = None
+    if "target_dtype_info" in node.meta:
+        bias_index = node.meta["target_dtype_info"].get("bias_index", None)
+    if bias_index is not None and bias_index < len(node.args) and node.args[bias_index] is arg:
+        return True
+    return node.kwargs.get("bias") is arg
```
Comment (jcaip) [inline torch/ao/quantization/fx/utils.py:105]: same here

**Review by jcaip — APPROVED** [review]
LGTM

File: torch/ao/quantization/fx/utils.py [hunk 2]
```diff
@@ -86,25 +86,23 @@ class ObservedGraphModuleAttrs:
...
-        if weight_index is not None and weight_index < len(node.args) and node.args[weight_index] is arg:
-            return True
-        return node.kwargs.get("weight") is arg
-    return False
+    weight_index = None
+    if "target_dtype_info" in node.meta:
+        weight_index = node.meta["target_dtype_info"].get("weight_index", None)
+    if weight_index is not None and weight_index < len(node.args) and node.args[weight_index] is arg:
+        return True
+    return node.kwargs.get("weight") is arg
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply torch/ao/quantization/fx/utils.py:96]: this might be needed for fx graph mode quantization

**Pull request closed without merging**